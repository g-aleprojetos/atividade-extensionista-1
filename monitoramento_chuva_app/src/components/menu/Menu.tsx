import React, {useState} from 'react';
import {TextoSmall, TextoXSmall} from 'components/texto';
import {useNavigate} from 'react-router-dom';
import rotas from 'resources/rotas';
import colors from 'resources/colors';
import * as S from './Menu.styles';
import {useAutenticadoContext} from 'context/autenticador';
import {Roles} from 'resources/interfaces';

export type Props = S.PropsMenu & {
  handleLogout: () => void;
};

export const Menu = (props: Props) => {
  const {handleLogout} = props;
  const navigate = useNavigate();
  const {usuario} = useAutenticadoContext();
  const [telaAtiva, setTelaAtiva] = useState<string>(rotas.Dashboard);

  const handleNavegate = (item: string) => {
    setTelaAtiva(item);
    navigate(item);
  };

  return (
    <S.ContainerNav data-testid={'test_menu'}>
      <S.ContainerContent data-testid={'test_containerContent'}>
        {usuario?.role === Roles.ADMIN && (
          <>
            <S.ItemNav
              isActive={telaAtiva === rotas.Dashboard}
              onClick={() => {
                handleNavegate(rotas.Dashboard);
              }}>
              {telaAtiva === rotas.Dashboard ? (
                <TextoSmall peso="bold" cursor="default">
                  Dashboard
                </TextoSmall>
              ) : (
                <TextoXSmall cursor="pointer" cor={colors.lightgray}>
                  Dashboard
                </TextoXSmall>
              )}
            </S.ItemNav>
            <S.ItemNav
              isActive={telaAtiva === rotas.CadastroUsuario}
              onClick={() => {
                handleNavegate(rotas.CadastroUsuario);
              }}>
              {telaAtiva === rotas.CadastroUsuario ? (
                <TextoSmall peso="bold" cursor="default">
                  Cadastrar Usuario
                </TextoSmall>
              ) : (
                <TextoXSmall cursor="pointer" cor={colors.lightgray}>
                  Cadastrar Usuario
                </TextoXSmall>
              )}
            </S.ItemNav>
            <S.ItemNav
              isActive={telaAtiva === rotas.CadastroDevice}
              onClick={() => {
                handleNavegate(rotas.CadastroDevice);
              }}>
              {telaAtiva === rotas.CadastroDevice ? (
                <TextoSmall peso="bold" cursor="default">
                  Cadastrar Device
                </TextoSmall>
              ) : (
                <TextoXSmall cursor="pointer" cor={colors.lightgray}>
                  Cadastrar Device
                </TextoXSmall>
              )}
            </S.ItemNav>
          </>
        )}
        <S.ItemNav
          onClick={() => {
            handleLogout();
          }}>
          <TextoXSmall cursor="pointer" cor={colors.lightgray}>
            Logout
          </TextoXSmall>
        </S.ItemNav>
      </S.ContainerContent>
    </S.ContainerNav>
  );
};
