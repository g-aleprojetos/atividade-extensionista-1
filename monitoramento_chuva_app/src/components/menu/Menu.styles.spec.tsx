import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as S from './Menu.styles';

describe('menu.styles', () => {
  test('ContainerNav DEVE ser igual ao snapshot', () => {
    const containerNav = renderer.create(<S.ContainerNav />).toJSON();
    expect(containerNav).toMatchSnapshot();
  });
  test('ContainerContent DEVE ser igual ao snapshot', () => {
    const containerContent = renderer
      .create(
        <S.ContainerContent />,
      )
      .toJSON();
    expect(containerContent).toMatchSnapshot();
  });

  test('Item DEVE ser igual ao snapshot.', () => {
    const item = renderer.create(<S.Item />).toJSON();
    expect(item).toMatchSnapshot();
  });
  test('ItemNav DEVE ser igual ao snapshot QUANDO "isActive" for false.', () => {
    const item = renderer.create(<S.Item isActive={false} />).toJSON();
    expect(item).toMatchSnapshot();
  });
  test('ItemNav DEVE ser igual ao snapshot QUANDO "isActive" for true.', () => {
    const item = renderer.create(<S.Item isActive={true} />).toJSON();
    expect(item).toMatchSnapshot();
  });
});
