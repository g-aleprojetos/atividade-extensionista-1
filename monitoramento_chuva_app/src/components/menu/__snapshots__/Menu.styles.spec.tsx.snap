// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu.styles ContainerContent DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  position: static;
  background-color: 'transparent';
  -webkit-transform: 'none';
  -ms-transform: 'none';
  transform: 'none';
  top: 'auto';
  right: 'auto';
  width: 'auto';
  height: auto;
  padding-top: 'auto';
  -webkit-transition: 'none';
  transition: 'none';
}

<ul
  className="c0"
/>
`;

exports[`menu.styles ContainerNav DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 5.563rem;
  padding-right: 3.438rem;
}

<nav
  className="c0"
/>
`;

exports[`menu.styles Item DEVE ser igual ao snapshot. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rempx;
  margin: 0.313rem 1.313rem;
  color: #ffffff;
}

<li
  className="c0"
/>
`;

exports[`menu.styles ItemNav DEVE ser igual ao snapshot QUANDO "isActive" for false. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rempx;
  margin: 0.313rem 1.313rem;
  color: #ffffff;
}

<li
  className="c0"
/>
`;

exports[`menu.styles ItemNav DEVE ser igual ao snapshot QUANDO "isActive" for true. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5rempx;
  margin: 0.313rem 1.313rem;
  color: #ffffff;
}

<li
  className="c0"
/>
`;
